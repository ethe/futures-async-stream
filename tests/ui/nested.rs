// SPDX-License-Identifier: Apache-2.0 OR MIT

#![feature(coroutines)]

use futures_async_stream::stream;

#[stream(item = i32)]
async fn stream(x: i32) {
    for i in 1..=x {
        yield i
    }
}

#[stream(item = i32)]
async fn _stream1() {
    async {
        #[for_await]
        for i in stream(2) {
            yield i * i; //~ ERROR `async` coroutines are not yet supported [E0727]
        }
    }
    .await;
}

fn main() {}
