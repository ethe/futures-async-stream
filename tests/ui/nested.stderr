error[E0727]: `async` coroutines are not yet supported
  --> tests/ui/nested.rs:19:13
   |
19 |             yield i * i; //~ ERROR `async` coroutines are not yet supported [E0727]
   |             ^^^^^^^^^^^

error[E0308]: mismatched types
  --> tests/ui/nested.rs:19:19
   |
19 |             yield i * i; //~ ERROR `async` coroutines are not yet supported [E0727]
   |                   ^^^^^ expected `()`, found `i32`
